include/aroma/types.h

```diff
@@ -48,4 +48,25 @@ typedef void * voidp;
 typedef uint32_t uchar;
 typedef uchar * ucharp;
 
+/* MIN / MAX */
+static inline int __MAX(int a, int b){
+  return (a>b)?a:b;
+}
+static inline int __MIN(int a, int b){
+  return (a<b)?a:b;
+}
+static inline float __FMAX(float a, float b){
+  return (a>b)?a:b;
+}
+static inline float __FMIN(float a, float b){
+  return (a<b)?a:b;
+}
+static inline double __DMAX(double a, double b){
+  return (a>b)?a:b;
+}
+static inline double __DMIN(double a, double b){
+  return (a<b)?a:b;
+}
```

Comment:
those names violate the C standard: http://port70.net/~nsz/c/c11/n1570.html#7.1.3
`All identifiers that begin with an underscore and either an uppercase letter or another underscore are always reserved for any use.`

Also, what's the point of e.g. calling it `__MAX` and then have a macro `MAX` alias it? Why not give it that name directly?
Also, `MIN` and `MAX` are defined in many headers and there's a reason non of them are implemented as C functions: so they work with all types.
While the prefixed DMIN/DMAX and FMIN/FMAX might be a good idea, MIN/MAX is not, since most people would expect them to work for all types.

Also, I'd suggest using lowercase names for functions because uppercase function names are generally accepted to be macros which users would expect to support all types.